jupyter_kernel_test/__init__.py

```diff
@@ -188,11 +204,111 @@ def test_display_data(self):
             raise SkipTest
 
         for sample in self.code_display_data:
-            self.flush_channels()
-            reply, output_msgs = self.execute_helper(sample['code'])
+            with self.subTest(code=sample['code']):
+                self.flush_channels()
+                reply, output_msgs = self.execute_helper(sample['code'])
+
+                self.assertEqual(reply['content']['status'], 'ok')
+
+                self.assertGreaterEqual(len(output_msgs), 1)
+                self.assertEqual(output_msgs[0]['msg_type'], 'display_data')
+                self.assertIn(sample['mime'], output_msgs[0]['content']['data'])
+
+    def test_silent(self):
+        self.flush_channels()
+
+        reply, output_msgs = self.execute_helper("", silent=True)
```

Comment:
The messaging protocol documentation implies this should be a supported operation

> In order to obtain the current execution counter for the purposes of displaying input prompts, frontends may make an execution request with an empty code string and `silent=True`.

but I take the point that this doesn't actually very well test the semantics, and that it would be better only to have tests with corresponding attributes. I'll see if I can think of a better approach for this one or drop it.